### PR TITLE
reflect-cpp: add version 0.13.0

### PR DESCRIPTION
**Pull request by toge**
### Summary
Changes to recipe:  **reflect-cpp/0.13.0**

#### Motivation
There are several new features in 0.13.0.

#### Details
https://github.com/getml/reflect-cpp/compare/v0.11.1...v0.13.0

---
- [x] Read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md)
- [x] Checked that this PR is not a duplicate: [list of PRs by recipe](https://github.com/conan-io/conan-center-index/discussions/24240)
- [x] Tested locally with at least one configuration using a recent version of Conan

### CODE REVIEW

**Review by AbrilRBS — COMMENTED** [review]
Note sure if this is an issue in the recipe or upstream, let me knowonce youy ifgure this out/if you need any help doing so @toge thanks!

**Comment by toge** [comment]
@AbrilRBS 
Thanks for your comment!
It seems to be an upstream issue.
I'm trying to reproduce it without conan.

File: recipes/reflect-cpp/all/conanfile.py [hunk 1]
```diff
@@ -78,6 +78,9 @@ def validate(self):
 
     def source(self):
         get(self, **self.conan_data["sources"][self.version], strip_root=True)
+        replace_in_file(self, os.path.join(self.source_folder, "include", "rfl", "parsing", "NamedTupleParser.hpp"),
```
[the remark this answers is not in the document]
Comment (toge) [reply recipes/reflect-cpp/all/conanfile.py]: @AbrilRBS 
This patch is made by me.
My MSVC environment is newer and does not have the errors in CCI's CI, so I used CCI's CI to figure out how to fix it.
It seems to compile successfully, so I will try to create a patch to upstream.

[the remark this answers is not in the document]
Comment (perseoGI) [reply recipes/reflect-cpp/all/conanfile.py]: Hi @toge!
Did you manage to open a PR upstream? https://github.com/getml/reflect-cpp/pulls?q=sort%3Aupdated-desc+is%3Apr+is%3Aopen

Thank you in advance!

[the remark this answers is not in the document]
Comment (toge) [reply recipes/reflect-cpp/all/conanfile.py]: @perseoGI 
I tried to create a PR for upstream, but the code for upstream has already changed too much to have it incorporate the differences from the 0.13.0 code base.
It looks like we will have to package 0.13.0 as a version without MSVC support or wait until the next release (0.14.0?).

**Comment by valgur** [comment]
https://github.com/getml/reflect-cpp/releases/tag/v0.14.0 was released yesterday

**Comment by AbrilRBS** [comment]
Pinging @toge, as @valgur mentioned, 0.14.0 has now released, we can package that version :)

**Comment by toge** [comment]
Closed.
New package with valid package names has been created.
https://github.com/conan-io/conan-center-index/pull/24857

**Pull request closed without merging**